fleet_adapter_r3/fleet_adapter_r3/RobotCommandHandle.py

```diff
@@ -274,10 +296,18 @@ def _follow_path():
 
                     self.node.get_logger().info(f'Attempting to navigate in map: {current_map_name}')
 
-                    target_pose_rmf = self.target_waypoint.position
-                    target_pose_robot: LionsbotCoord = self.transforms[current_map_name].rmf_meters_to_robot(
-                        rmf_coord=target_pose_rmf
-                    )
+                    if self.use_safe_nav:
+                        # Search through all waypoints to get matching waypoint.
+                        # Once found, make API call to get LionsbotCoord of matching waypoint.
+                        for waypoint_name in self.waypoint_network:
+                            if (self.target_waypoint.position.x == self.waypoint_network[waypoint_name][0] and
+                                self.target_waypoint.position.y == self.waypoint_network[waypoint_name][1]):
+                                target_pose_robot = self.api.get_waypoint_position(self.name, self.map_name, waypoint_name)
```

Comment:
question: Would an inaccurate transform cause problems here? Just worried if the x and y doesn't match with what is stored in the robot

Reply:
As long as `self.target_waypoint.position.x` and `self.target_waypoint.position.y` matches with the waypoint defined in RMF Map, it will be able to get the robot position via its matching location point name on the robot. 

The only thing that needs to match when stored on the robot is the waypoint name.

```bash
RMF Map Waypoint Name = Robot Map Waypoint Name
```

Here's a psuedocode for explaining better what this code segment does:

```py
# Search through all RMF Waypoint
    # If Input RMF Waypoint (x,y) matches Existing RMF Waypoint (x,y)
        # Use RMF Waypoint Name to get the corresponding Location Point (x,y) in LionsBotCoord from Robot 
```

Will improve the comment here for better clarity, going forward.